op-node/p2p/cli/load_config.go

```diff
@@ -0,0 +1,301 @@
+	}
+
+	p, err := loadNetworkPrivKey(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to load p2p priv key: %w", err)
+	}
+	conf.Priv = p
+
+	if err := loadListenOpts(conf, ctx); err != nil {
+		return nil, fmt.Errorf("failed to load p2p listen options: %w", err)
+	}
+
+	if err := loadDiscoveryOpts(conf, ctx); err != nil {
+		return nil, fmt.Errorf("failed to load p2p discovery options: %w", err)
+	}
+
+	if err := loadLibp2pOpts(conf, ctx); err != nil {
+		return nil, fmt.Errorf("failed to load p2p options: %w", err)
+	}
+
+	if err := loadGossipOptions(conf, ctx); err != nil {
+		return nil, fmt.Errorf("failed to load p2p gossip options: %w", err)
+	}
+
+	conf.ConnGater = p2p.DefaultConnGater
+	conf.ConnMngr = p2p.DefaultConnManager
+
+	return conf, nil
+}
+
+func validatePort(p uint) (uint16, error) {
+	if p == 0 {
+		return 0, nil
+	}
+	if p >= (1 << 16) {
+		return 0, fmt.Errorf("port out of range: %d", p)
+	}
+	if p < 1024 {
+		return 0, fmt.Errorf("port is reserved for system: %d", p)
+	}
+	return uint16(p), nil
+}
+
+func loadListenOpts(conf *p2p.Config, ctx *cli.Context) error {
+	listenIP := ctx.GlobalString(flags.ListenIP.Name)
+	if listenIP != "" { // optional
+		conf.ListenIP = net.ParseIP(listenIP)
+		if conf.ListenIP == nil {
+			return fmt.Errorf("failed to parse IP %q", listenIP)
+		}
+	}
+	var err error
+	conf.ListenTCPPort, err = validatePort(ctx.GlobalUint(flags.ListenTCPPort.Name))
+	if err != nil {
+		return fmt.Errorf("bad listen TCP port: %w", err)
+	}
+	conf.ListenUDPPort, err = validatePort(ctx.GlobalUint(flags.ListenUDPPort.Name))
+	if err != nil {
+		return fmt.Errorf("bad listen UDP port: %w", err)
+	}
+	return nil
+}
+
+func loadDiscoveryOpts(conf *p2p.Config, ctx *cli.Context) error {
+	if ctx.GlobalBool(flags.NoDiscovery.Name) {
+		conf.NoDiscovery = true
+	}
+
+	var err error
+	conf.AdvertiseTCPPort, err = validatePort(ctx.GlobalUint(flags.AdvertiseTCPPort.Name))
+	if err != nil {
+		return fmt.Errorf("bad advertised TCP port: %w", err)
+	}
+	conf.AdvertiseUDPPort, err = validatePort(ctx.GlobalUint(flags.AdvertiseUDPPort.Name))
+	if err != nil {
+		return fmt.Errorf("bad advertised UDP port: %w", err)
+	}
+	adIP := ctx.GlobalString(flags.AdvertiseIP.Name)
+	if adIP != "" { // optional
+		ips, err := net.LookupIP(adIP)
+		if err != nil {
+			return fmt.Errorf("failed to lookup IP of %q to advertise in ENR: %w", adIP, err)
+		}
+		// Find the first v4 IP it resolves to
+		for _, ip := range ips {
+			if ipv4 := ip.To4(); ipv4 != nil {
+				conf.AdvertiseIP = ipv4
+				break
+			}
+		}
+		if conf.AdvertiseIP == nil {
+			return fmt.Errorf("failed to parse IP %q", adIP)
+		}
+	}
+
+	dbPath := ctx.GlobalString(flags.DiscoveryPath.Name)
+	if dbPath == "" {
+		dbPath = "opnode_discovery_db"
+	}
+	if dbPath == "memory" {
+		dbPath = ""
+	}
+	conf.DiscoveryDB, err = enode.OpenDB(dbPath)
+	if err != nil {
+		return fmt.Errorf("failed to open discovery db: %w", err)
+	}
+
+	conf.Bootnodes = p2p.DefaultBootnodes
```

Comment:
Are there any cases in which users would not want to use the default boot nodes?

Reply:
ie whoever runs the default boot nodes gets IPs of every node on the network right?

Reply:
Didn't put the default likes this in the PR (just moved the code), but I think it makes sense from a UX perspective. Agreed we could make it fully override-able. Want me to change it to be the default value, and get wiped if you specify another value? @mslipper might rely on current behavior.